domain/miningmanager/mempool/orphan_pool.go

```diff
@@ -15,47 +15,102 @@ import (
 	"github.com/pkg/errors"
 )
 
-type idToOrphan map[externalapi.DomainTransactionID]*model.OrphanTransaction
-type previousOutpointToOrphans map[externalapi.DomainOutpoint]idToOrphan
+type idsToOrphans map[externalapi.DomainTransactionID]*model.OrphanTransaction
+type previousOutpointToOrphans map[externalapi.DomainOutpoint]*model.OrphanTransaction
 
 type orphansPool struct {
 	mempool                   *mempool
-	allOrphans                idToOrphan
+	allOrphans                idsToOrphans
 	orphansByPreviousOutpoint previousOutpointToOrphans
 	lastExpireScan            uint64
 }
 
 func newOrphansPool(mp *mempool) *orphansPool {
 	return &orphansPool{
 		mempool:                   mp,
-		allOrphans:                idToOrphan{},
+		allOrphans:                idsToOrphans{},
 		orphansByPreviousOutpoint: previousOutpointToOrphans{},
 		lastExpireScan:            0,
 	}
 }
 
 // this function MUST be called with the mempool mutex locked for writes
 func (op *orphansPool) maybeAddOrphan(transaction *externalapi.DomainTransaction, isHighPriority bool) error {
+	err := op.checkOrphanDuplicate(transaction)
+	if err != nil {
+		return err
+	}
+
+	err = op.checkOrphanSize(transaction)
+	if err != nil {
+		return err
+	}
+	if op.mempool.config.MaximumOrphanTransactionCount <= 0 {
```

Comment:
For the record: I thought so as well, but decided to not argue with previous design about this (figuring that they probably wanted to make sure that orphans are rejected over size even if we don't accept orphans).

But yeah, I agree with you, and since we both agree, doing as suggested.